var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -110,11 +110,10 @@ class Cmake(Package):
 
     phases = ['bootstrap', 'build', 'install']
 
-    def setup_environment(self, spack_env, run_env):
-        if self.compiler.name == 'fj' \
-                and self.compiler.cxx11_flag \
-                not in self.spec.compiler_flags['cxxflags']:
-            spack_env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)
+    def flag_handler(self, name, flags):
+        if name == 'cxxflags' and self.compiler.name == 'fc':
+            flags.append(self.compiler.cxx11_flag)
```

Comment:
What if the user tried to build with other possible cxx flags?  Is it going to raise an error when it sees `cxx14_flag` and `cxx11_flag` in there?  Also what if `cxx11_flag` is already in flags?

Reply:
@tgamblin @t-karatsu @chuckatkins I've added a commit to do some error checking in the flag handler. I've designed it to be easy to extend when future standards flags are added to the fj compiler going forward. Let me know what you think

Reply:
lgtm but given that i jumped the gun on the merge last time I'll leave it to someone else to give it the green light.

Reply:
I think that the correction is good and easy to understand. Thank you for making it better.